src/mcts/search.cc

```diff
@@ -540,42 +544,44 @@ void Search::EnsureBestMoveKnown() REQUIRES(nodes_mutex_)
 }
 
 // Returns @count children with most visits.
-std::vector<EdgeAndNode> Search::GetBestChildrenNoTemperature(Node* parent,
-                                                              int count) const {
+std::vector<Search::TerminalScore> Search::GetBestChildrenNoTemperature(
+    Node* parent, int count) const {
   MoveList root_limit;
   if (parent == root_node_) {
     PopulateRootMoveLimit(&root_limit);
   }
   // Best child is selected using the following criteria:
+  // * Best terminal score (favor wins, avoid losses).
   // * Largest number of playouts.
   // * If two nodes have equal number:
   //   * If that number is 0, the one with larger prior wins.
   //   * If that number is larger than 0, the one with larger eval wins.
-  using El = std::tuple<uint64_t, float, float, EdgeAndNode>;
+  using El = std::tuple<TerminalScore, uint64_t, float, float>;
```

Comment:
Maybe instead of doing that, add `int GetMovesTillCheckmate()` to `EdgeAndNode`, add that as the first element to the tuple, and leave the rest as it is.
And then when priting info, printing function will have to call that again... Which maybe not that bad as usually there's no multipv or something.

Reply:
A plain `int` ordering of moves-till-checkmate probably isn't sufficient as we want to prefer:
1) shorter winning mates: GMTC() = 1
2) longer winning mates: GMTC() = 12
3) shorter TB win: GMTC() = 1000
4) longer TB win: GMTC() = 1234
5) not terminal win/loss: GMTC() = 0 ?
6) longer TB loss: GMTC() = -1234
7) shorter TB loss: GMTC() = -1000
8) longer losing mates: GMTC() = -12
9) shorter losing mates: GMTC() = -1

We could instead emplace back a transformed GMTC() with some magic number and formula for sorting purposes… ?

Reply:
Oh, I suppose `emplace_back(1.0f / GMTC(), …)` might work…

Reply:
0 needs to be handled separately anyway. So maybe to keep it an int something like `if < 0: intmin - gmtc; if > 0: intmax - gmtc`

Reply:
GMTC() could return floats instead including `std::numeric_limits<float>::infinity()` for not terminal win/loss… although that makes other callers, e.g., for reporting mates, slightly more complicated.

Reply:
Ended up adding two EdgeAndNode helpers: `GetMovesTillCheckmate()` and `GetTerminalSortingKey()`

Removed the class